pkg/products/rhssouser/reconciler.go

```diff
@@ -225,9 +245,334 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, installation *inte
 	}
 	r.logger.Infof("The operation result for keycloakrealm %s was %s", kcr.Name, or)
 
+	// We want to update the master realm by adding an openshift-v4 idp. We can not add the idp until we know the host
+	// url - kc.Status.InternalURL
+	if kc.Status.InternalURL != "" {
+		host := kc.Status.InternalURL
+
+		// Create the master realm. The master real already exists in Keycloak but we need to get a reference to it
+		// in order to create the IDP and admin users on it
+		masterKcr, err := r.updateMasterRealm(ctx, serverClient, installation, host)
+		if err != nil {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+
+		kcClient, err := r.keycloakClientFactory.AuthenticatedClient(*kc)
+		if err != nil {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+
+		// Ensure the IDP exists before trying to create via rhsso client.
+		// We have to create via rhsso client as keycloak will not accept changes to the master realm, via cr changes,
+		// after its initial creation
+		if masterKcr.Spec.Realm.IdentityProviders != nil && masterKcr.Spec.Realm.IdentityProviders[0] != nil {
```

Comment:
same here what if we switched this to be  ``` if masterKcr.Spec.Realm.IdentityProviders == nil || masterKcr.Spec.Realm.IdentityProviders[0] == nil {  return }```

Reply:
ok, will do